Fix regression: use exact matches to find coincident vertices with topology edition

Djedouas opened the pull request:
## Description

Fixes #63305.

PR https://github.com/qgis/QGIS/pull/58352 introduced a regression not caught in review.

Finding coincident vertices for topology edition should not use a search radius, but an exact match.

Djedouas committed:
fix(qgsvertextool): use exact matches to find coincident vertices

nyalldawson commented:
@Djedouas could a test be introduced to "lock in" this desired behaviour? I feel like there's frequently a lot of regressions in topology editing, so expanding the test suite would definitely be valuable.

Djedouas committed:
fix(qgsvertextool): add test on coincident vertices with topological editing

Add a feature in line layer with a vertex near a polygon layer feature
vertex (at 4e-7 distance).
Verify that with the topological editing, moving the polygon vertex does
not move the line vertex because coincindent vertices must be with exact
coordinates (within 1e-8).

agiudiceandrea commented:
Since this PR fixes a regression, I think it should be directly backported to release-3_40 branch instead of the queued_ltr_backports branch.
